defold-simpledata/editor/src/simpledata-ext.clj

```diff
@@ -151,17 +209,26 @@
 
   ;; Outputs for internal use.
   (output simpledata-pb g/Any produce-simpledata-pb)
-  (output build-errors g/Any produce-build-errors)
 
   ;; Outputs we're expected to implement.
   (output form-data g/Any :cached produce-form-data)
-  (output node-outline outline/OutlineData :cached produce-node-outline)
   (output save-value g/Any (gu/passthrough simpledata-pb))
+  (output own-build-errors g/Any produce-own-build-errors)
   (output build-targets g/Any :cached produce-build-targets))
 
 ;;//////////////////////////////////////////////////////////////////////////////////////////////
 
-(defn- register-resource-types [workspace]
+(defn- register-resource-types
+  "Register our .simpledata resource type with the workspace. Whenever we find
+  a .simpledata file in the project, a SimpleDataNode is created for it. Then,
+  the load-fn populates the SimpleDataNode from a Clojure map representation of
+  the protobuf data we load from the .simpledata resource. When we register our
+  resource type, we also tag ourselves as a component that can be used in game
+  objects, and declare which view types can present our resource for editing in
+  an editor tab. In our case, we will use the form view for editing. To work
+  with the form view, our node is expected to implement the form-data output
+  required by the form view."
+  [workspace]
```

Comment:
Can we move the comments out of the functions?
It doesn't have to be an idiomatic comment on the function?

RIght now, the comment is longer than the function itself, while at the same time splits the function into two parts, which imho makes it much less readable.

Also, for longer texts, feel free to add a new readme file.